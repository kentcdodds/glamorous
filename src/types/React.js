// @flow
import type {Component, Element} from 'react'

export type FunctionalComponent = (
  {[key: string]: any}, // props
  ?{[key: string]: any}, // context
) => Element<*> | null;

// eslint-disable-next-line no-undef
export type ComponentClass = Class<Component<*, *, *>>;

export type RefCallback<Instance> = (Instance | null) => void;
