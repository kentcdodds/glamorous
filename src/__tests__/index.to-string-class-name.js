/* eslint func-style:0, react/prop-types:0 */
import React from 'react'
import {render} from 'enzyme'
import * as jestGlamorReact from 'jest-glamor-react'
import glamorous from '../'

expect.extend(jestGlamorReact.matcher)
expect.addSnapshotSerializer(jestGlamorReact.serializer)

beforeEach(() => {
  glamorous.config.idCounter = 1
})

test('allows you to use a component as a child selector', () => {
  const RedDiv = glamorous.div({color: 'red'})
  const ContainerDiv = glamorous.div({
    [RedDiv]: {
      color: 'blue',
    },
  })

  expect(
    render(<ContainerDiv><RedDiv>Hello</RedDiv></ContainerDiv>),
  ).toMatchSnapshotWithGlamor()
})

test('allows you to use a component as any kind of selector', () => {
  const ContainerDiv = glamorous.div({
    color: 'blue',
  })
  const RedDiv = glamorous.div(
    {
      [`${ContainerDiv} &`]: {
        color: 'red',
      },
    },
    {
      [`${ContainerDiv} &`]: {
        color: 'green',
      },
    },
  )

  expect(
    render(<ContainerDiv><RedDiv>Hello</RedDiv></ContainerDiv>),
  ).toMatchSnapshotWithGlamor()
})
