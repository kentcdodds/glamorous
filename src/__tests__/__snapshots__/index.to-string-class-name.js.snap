// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allows you to use a component as a child selector 1`] = `
.css-3v2xme .g-1-1,
[data-css-3v2xme] .g-1-1 {
  color: blue;
}

.css-1ezp9xe,
[data-css-1ezp9xe] {
  color: red;
}

<div
  class="css-3v2xme"
>
  <div
    class="css-1ezp9xe g-1-1"
  >
    Hello
  </div>
</div>
`;

exports[`allows you to use a component as any kind of selector 1`] = `
.css-icjsl7,
[data-css-icjsl7] {
  color: blue;
}

.g-1-1 .css-1ha6yxt,
.g-1-1 [data-css-1ha6yxt] {
  color: green;
}

<div
  class="css-icjsl7 g-1-1"
>
  <div
    class="css-1ha6yxt"
  >
    Hello
  </div>
</div>
`;
